### PR TITLE
Update the CLI Use Cases (Part 2 of CLI revisions)

### PR DESCRIPTION
**Pull request by jmintuitive**
### Description

This updates the "Use Case" section of the CLI docs. It's a bit more involved than the previous section as many more pages linked to the old docs that these are replacing. There are also just more pages involved in this one. 

Specifically this PR:
1. Updates all pages as written in the detailed drafts.
2. Updates `docusaurus.ts` with the redirects needed so old page links still work.
3. Updates all relative links within the docs to go to the proper new pages.
4. Updates `sidebars.ts` to have the proper navigation.

### Checklist

- Do all Lints pass?
  - [] Have you ran `pnpm spellcheck`? -> No, I was unable to run this command still.
  - [X] Have you ran `pnpm fmt`?
  - [X] Have you ran `pnpm lint`?

### CODE REVIEW

**Comment by jmintuitive** [comment]
Note: This is a draft PR as there are still a couple reviews / clean up things I have to do before this is ready to merge :) 

Specifically: 
1. I need to verify each page's copy / styling looks good in the local preview. 
2. I need to add more redirects from old pages to the new ones in the `docusaurus.ts`.
3. I need to fix a small issue where when trying to use the same page in 2 spots on the sidebar, it will jump to the "highest" when actually on the page. (Likely I'll just need to create a stub in one location that links to the actual page in the other spot). 

When I'm done with that I'll take this out of draft and ping the reviewers :)

File: apps/docusaurus/docs/tools/aptos-cli/use-cli/index.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# Use the Aptos CLI
+
+The Aptos command line interface (CLI) is a tool to help you compile and test Move contracts and quickly play with Aptos features on-chain.
+
+You can explore each of the use-cases here:
+
+1. [Working with Move Contracts](https://www.notion.so/Working-With-Move-Contracts-48db6aea8bf54a3f88dd9de06cdd1587?pvs=21)
+2. [Trying things on-chain](https://www.notion.so/Trying-Things-On-Chain-With-Aptos-CLI-5cba791bdc9a4ee9ae31f65fc135c065?pvs=21)
+3. [Running a local Aptos network](https://www.notion.so/Running-a-Local-Network-via-Aptos-CLI-f1b557c57150491f8eb2dde51d6b66c8?pvs=21)
+4. [Managing a network node](https://www.notion.so/Aptos-CLI-Docs-Revised-Outline-5fbf805486fe43dc909bba14c5ee95db?pvs=21)
```
[the remark this answers is not in the document]
Comment (jmintuitive) [reply apps/docusaurus/docs/tools/aptos-cli/use-cli/index.md:14]: Oops, yes those are supposed to be relative links. Fixed :)

File: apps/docusaurus/docs/guides/system-integrators-guide.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ See [Aptos Blockchain Networks](../nodes/networks.md) for full details on each e
 
 There are two options for running a local testnet:
 
-- [Install the Aptos CLI](../tools/aptos-cli/install-cli/index.md) and 2) run a [local development network](./local-development-network.md). This path is useful for developing on the Aptos blockchain, debugging Move contracts, and testing node operations. Using the CLI you will have a fully featured local development environment including a single node network, the node API, indexer API, and a faucet.
+- [Install the Aptos CLI](../tools/aptos-cli/install-cli/index.md) and 2) run a [local development network](../tools/aptos-cli/use-cli/running-a-local-network.md). This path is useful for developing on the Aptos blockchain, debugging Move contracts, and testing node operations. Using the CLI you will have a fully featured local development environment including a single node network, the node API, indexer API, and a faucet.
```
[the remark this answers is not in the document]
Comment (jmintuitive) [reply apps/docusaurus/docs/guides/system-integrators-guide.md:44]: Reworded :)

File: apps/docusaurus/docs/guides/system-integrators-guide.md [hunk 3]
```diff
@@ -86,7 +86,7 @@ Aptos currently provides three SDKs:
 2. [Python](../sdks/python-sdk/index.md)
 3. [Rust](../sdks/rust-sdk/index.md)
 
-Almost all developers will benefit from exploring the CLI. [Using the CLI](../tools/aptos-cli/use-cli/use-aptos-cli.md) demonstrates how the CLI can be used to which includes creating accounts, transferring coins, and publishing modules.
+Almost all developers will benefit from exploring the CLI. [Using the CLI](../tools/aptos-cli/use-cli/index.md) demonstrates how the CLI can be used to which includes creating accounts, transferring coins, and publishing modules.
```
[the remark this answers is not in the document]
Comment (jmintuitive) [reply apps/docusaurus/docs/guides/system-integrators-guide.md]: Updated this sentence as well :)

**Comment by jmintuitive** [comment]
@gregnazario @0xZihan This PR is now ready for review :)

**Comment by jmintuitive** [comment]
There was one issue I wasn't sure the best way to address. 

Two sidebar paths go to the same content because it fits both places. Specifically:

1. Advanced Builder Guides > Develop Locally > Running A Local Network
2. Aptos CLI > Use Aptos CLI > Running A Local Network

Both go to the equivalent of this page: https://aptos.dev/guides/local-development-network

The problem is that when you load that page, the sidebar always shows it as the Advanced Builder Guide path. This can be jarring if you're clicking through the sidebar items in the Use Aptos CLI section as you're suddenly in a different menu. This is not really noticeable if you find the page through on-page links like the `Use Aptos CLI` page. 

Options:
1. I could add a stub page to one of the two spots that links to the other. 
2. I could copy the content into another file in the other path (I'm averse to that because it is odd to have 2 identical pages to update, easy to fall out of sync). 
3. We could say this is ok
4. Something I haven't thought of :)

Any thoughts on what would be best?

File: apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md [hunk 4]
```diff
@@ -9,44 +9,45 @@ If you are using the CLI to try things out on-chain, you will need to configure
...
 ## (Optional) Creating Named Configurations (Profiles)
 
-For testing more complicated scenarios, you will often want multiple accounts on-chain. One way to do this is to create a named configuration which we will call a profile.
+For testing more complicated scenarios, you will often want multiple accounts on-chain. One way to do this is to create a named configuration which we call a profile.
 
 To create a profile, run `aptos init --profile <name_of_profile>`. The configuration you generate will be usable when calling CLI commands as replacements for arguments.
 
 For example:
-
+```bash
```
Comment (hariria) [inline apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md:50]: Same here

File: apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md [hunk 5]
```diff
@@ -63,25 +64,25 @@ One quality of life feature you can enable is shell auto-completions.
 1. Determine which shell you are using (you can run `echo $SHELL` if you are unsure).
 2. Look up where configuration files for shell completions go for that shell (it varies from shell to shell). The supported shells are `[bash, zsh, fish, PowerShell, elvish]`.
 3. Run the following command with your specific shell and the output file for completions using your shell:
-
+```bash
```
Comment (hariria) [inline apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md:67]: Same here

File: apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md [hunk 6]
```diff
@@ -63,25 +64,25 @@ One quality of life feature you can enable is shell auto-completions.
...
 ```zsh
 aptos config generate-shell-completions --shell <YOUR_SHELL_HERE> --output-file <OUTPUT_DESTINATION_FOR_YOUR_SHELL>
 ```
 
 Example command for [`oh my zsh`](https://ohmyz.sh/):
-
+```bash
```
Comment (hariria) [inline apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md:73]: Same here

File: apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md [hunk 7]
```diff
@@ -63,25 +64,25 @@ One quality of life feature you can enable is shell auto-completions.
...
 
-By default, the configuration is set to use a local file (`.aptos/config.yaml`) per workspace. If you would like to use one shared configuration for all workspaces, you can follow these steps.
+By default, the CLI will look for a configuration in `.aptos/config.yaml` in each workspace directory. If you would like to use a shared configuration for all workspaces, you can follow these steps:
 
-1. Create a folder in your home directory called `.aptos` (it will have the path `~/.aptos`).
+1. Create a folder in your home directory called `.aptos` (so it has the path `~/.aptos`).
 2. Create a yaml file inside `.aptos` called `global_config.yaml`.
 3. Run the command:
-
+```bash
```
Comment (hariria) [inline apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md:85]: Same here

File: apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md [hunk 8]
```diff
@@ -9,44 +9,45 @@ If you are using the CLI to try things out on-chain, you will need to configure
...
 :::
 
 1. Run `aptos init` and follow the instructions in the command line.
-   1. To use default settings, you can provide no input and just press “Enter”.
 
-```zsh
-aptos init
+   1. To use default settings, you can provide no input and just press “Enter”. For example:
+```bash
+   ```zsh
```
[the remark this answers is not in the document]
Comment (jmintuitive) [reply apps/docusaurus/docs/tools/aptos-cli/cli-configuration.md]: They are not, that's super weird. I did a search and replace from ```zsh to ```bash, so no clue how that happened. Fixed now though :)

**Comment by gregnazario** [comment]
> 
> There was one issue I wasn't sure the best way to address. 
> 
> Two sidebar paths go to the same content because it fits both places. Specifically:
> 
> 1. Advanced Builder Guides > Develop Locally > Running A Local Network
> 2. Aptos CLI > Use Aptos CLI > Running A Local Network
> 
> Both go to the equivalent of this page: https://aptos.dev/guides/local-development-network
> 
> The problem is that when you load that page, the sidebar always shows it as the Advanced Builder Guide path. This can be jarring if you're clicking through the sidebar items in the Use Aptos CLI section as you're suddenly in a different menu. This is not really noticeable if you find the page through on-page links like the `Use Aptos CLI` page. 
> 
> Options:
> 1. I could add a stub page to one of the two spots that links to the other. 
> 2. I could copy the content into another file in the other path (I'm averse to that because it is odd to have 2 identical pages to update, easy to fall out of sync). 
> 3. We could say this is ok
> 4. Something I haven't thought of :)
> 
> Any thoughts on what would be best

Just drop the advanced builder page one

**Comment by jmintuitive** [comment]
> Just drop the advanced builder page one

Done :)

**Review by hariria — APPROVED** [review]
LGTM will let @gregnazario take another pass

**Review by 0xZihan — APPROVED** [review]

**Pull request merged**